internal/characters/thoma/skill.go

```diff
@@ -53,18 +58,20 @@ func (c *char) Skill(p map[string]int) action.ActionInfo {
 	if !ok {
 		panic("target 0 should be Player but is not!!")
 	}
-	player.ApplySelfInfusion(attributes.Pyro, 25, 30)
+	c.QueueCharTask(func() {
+		player.ApplySelfInfusion(attributes.Pyro, 25, 30)
+	}, skillHitmark)
```

Comment:
self infusion starts on cooldown start (9), not hitmark, also don't really need char queue for this